enhancements/automated-resource-request-scaling.md

```diff
@@ -247,10 +151,55 @@ their own workloads. OpenShift currently requires 7.
 
 ### Implementation Details/Notes/Constraints
 
-1. Add the new field to the `Infrastructure` API.
-2. Expose the `controlPlaneCPULimit` setting through the installer via
-   `install-config.yaml`.
-3. Update operators to respond to the API, as described above.
+The `ClusterResourceOverride` API currently scales CPU request value
+as a percentage of the limit value. Our cluster operators and their
+operands are [deployed without limits
+set](https://github.com/openshift/enhancements/blob/master/CONVENTIONS.md#resources-and-limits),
+so we will need to add a new field to `ClusterResourceOverride` to
+describe how to scale in a different way.
+
+Of the [options identified for expressing the scaling
+factor](#expressing-a-need-to-scale-using-an-api), a scaling ratio
+would be the most consistent with the other fields already in the
+`ClusterResourceOverride` API and used by the admission webhook. It
+has the downside that a user needs to understand the base value so
+they can compute an appropriate ratio, but by using a ratio we can
+avoid encoding knowledge of the base configuration of OpenShift into
+an add-on operator that is released independently of OpenShift.
+
+Because the admission webhook is added after the cluster components
+are already running and it does not change existing Pods, we need
+something to recognize the Pods that need to be recreated so the
+scaling factor can be applied. A controller in the
+`cluster-resource-overide-admission` component can do this. It can
+coordinate with the admission webhook by looking for an annotation or
+label on the Pod to indicate that it has been scaled. Any Pods without
+the annotation in a namespace where scaling is enabled can be killed
+so they are recreated and the admission webhook can modify their
+settings.
+
+An additional benefit of using the separate component is that it is
+not part of the OpenShift release payload, and therefore could be
+released independently of OpenShift releases. Since we are looking at
+a tight timeline for delivering the work described in this
+enhancement, that may help with scheduling.
+
+1. Add a new field to `ClusterResourceOverride` API with a name like
```

Comment:
For a general case where a pod has a limit set, would need to decide the precendence if both cpuRequestPercent and cpuRequestToLimitPercent are set.

Reply:
Good point. It seems like the limit-based scaling (the existing behavior) should take precedence, and the new field should be ignored if the old field is present. What do you think?

Reply:
Agreed.